src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,46 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.service.producer.BulldozerProducer;
+import core.mate.academy.service.producer.ExcavatorProducer;
+import core.mate.academy.service.producer.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+
+    @Override
+    public List<T> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            MachineProducer<Bulldozer> machineList = new BulldozerProducer();
+            return (List<T>) machineList.get();
```

Comment:
```suggestion
            return (List<T>) bulldozerProducer.get();
```
you can create global variables for producers and only invoke them here